### PR TITLE
feat: Add instrumentation scope for metrics

### PR DESCRIPTION
**Pull request by rakesh-garimella**
## Description

Changes proposed in this pull request (what was done and why):

- set instrumentation scope when empty or already set
      -  `otelcol/kubeletstatsreceiver` -> `io.kyma-project.telemetry/runtime`
      -   `otelcol/prometheusreceiver` -> `io.kyma-project.telemetry/prometheus`
      -    `otelcol/prometheusreceiver` -> `io.kyma-project.telemetry/istio`
- remove `kyma.source` from agent 
- remove filtering of `kyma.source` in gateway
- Move from `datapoints` to `metrics` for filtering as its more efficient
- e2e tests will be implemented in following PR
Changes refer to particular issues, PRs or documents:

- https://github.com/kyma-project/telemetry-manager/issues/973

## Traceability
- [x] The PR is linked to a GitHub issue.
- [ ] New features have a milestone set.
- [ ] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [x] The corresponding GitHub issue has a respective `area` and `kind` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [ ] Adjusted the documentation if the change is user-facing.
- [x] The feature is unit-tested
- [ ] The feature is e2e-tested

### CODE REVIEW

File: internal/otelcollector/config/metric/agent/set_instrumention_scope.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+func makeInstrumentationStatement(inputSource metric.InputSourceType) string {
+	var instrumentationScopeName = map[metric.InputSourceType]string{
+		metric.InputSourceRuntime:    "otelcol/kubeletstatsreceiver",
+		metric.InputSourcePrometheus: "otelcol/prometheusreceiver",
+		metric.InputSourceIstio:      "otelcol/prometheusreceiver",
+	}
+
+	var transformedInstrumentationScopeName = map[metric.InputSourceType]string{
+		metric.InputSourceRuntime:    "io.kyma-project.telemetry/kubeletstatsreceiver",
```
Comment (a-thaler) [inline internal/otelcollector/config/metric/agent/set_instrumention_scope.go:35]: why kubeletstatsreceiver? The goal is to use the input name!

Comment (rakesh-garimella) [reply internal/otelcollector/config/metric/agent/set_instrumention_scope.go]: it has been changed

File: internal/otelcollector/config/metric/agent/set_instrumentation_scope_test.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
+package agent
+
+import (
+	"testing"
+
+	"github.com/kyma-project/telemetry-manager/internal/otelcollector/config"
+	"github.com/kyma-project/telemetry-manager/internal/otelcollector/config/metric"
+)
+
+func Test_transformedInstrumentationScope(t *testing.T) {
```
Comment (chrkl) [inline internal/otelcollector/config/metric/agent/set_instrumentation_scope_test.go:10]: Rename to TestTransformInstrumentationScope

File: internal/otelcollector/config/metric/agent/set_instrumention_scope.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+)
+
+func transformInstrumentationScope(inputSource metric.InputSourceType) *TransformProcessor {
+	return &TransformProcessor{
+		ErrorMode:        "ignore",
+		MetricStatements: makeMetricStatement(inputSource),
+	}
+}
+
+func makeMetricStatement(inputSource metric.InputSourceType) []config.TransformProcessorStatements {
```
Comment (skhalash) [inline internal/otelcollector/config/metric/agent/set_instrumention_scope.go:17]: Do we need this function?

File: internal/otelcollector/config/metric/agent/set_instrumention_scope.go [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package agent
+
+import (
+	"fmt"
+
+	"github.com/kyma-project/telemetry-manager/internal/otelcollector/config"
+	"github.com/kyma-project/telemetry-manager/internal/otelcollector/config/metric"
+)
+
+func transformInstrumentationScope(inputSource metric.InputSourceType) *TransformProcessor {
```
Comment (skhalash) [inline internal/otelcollector/config/metric/agent/set_instrumention_scope.go:10]: This func does not transform anything
```suggestion
func makeInstrumentationScopeProcessor(inputSource metric.InputSourceType) *TransformProcessor {
```

File: internal/otelcollector/config/metric/shared_attributes.go [hunk 5]
```diff
@@ -12,3 +8,21 @@ const (
...
+	TransformedInstrumentationScopeIstio      = "io.kyma-project.telemetry/istio"
+)
+
+var TransformedInstrumentationScope = map[InputSourceType]string{
+	InputSourceRuntime:    TransformedInstrumentationScopeRuntime,
+	InputSourcePrometheus: TransformedInstrumentationScopePrometheus,
+	InputSourceIstio:      TransformedInstrumentationScopeIstio,
+}
+
+var UpstreamInstrumentationScopeName = map[InputSourceType]string{
```
Comment (skhalash) [inline internal/otelcollector/config/metric/shared_attributes.go:24]: Does it have to be here? Isn't it an internal detail, only the metric agent should know about?

Comment (skhalash) [reply internal/otelcollector/config/metric/shared_attributes.go:24]: If you move it there, you could simplify the other constants:
TransformedInstrumentationScopeRuntime -> InstrumentationScopeRuntime

**Review by skhalash — CHANGES_REQUESTED** [review]

File: internal/otelcollector/config/metric/shared_attributes.go (hunk 5 above)
Comment (rakesh-garimella) [reply internal/otelcollector/config/metric/shared_attributes.go]: fixed

**Review by skhalash — APPROVED** [review]

**Pull request merged**